---
title: "R Notebook"
output: html_notebook
editor_options: 
  markdown: 
    wrap: 72
  chunk_output_type: console
---

# Little Greenhouse Experiment on the Prairie

This greenhouse experiment tests the influence of a plant species
on the rhizosphere community of a subsequently grown plant species. The
species tested were sunflower, pea, soybean, and maize on maize
rhizosphere communities and spring wheat, maize, soybean on soybean
rhizosphere communities

```{r, message = FALSE, echo=FALSE}
library(vegan)
library(pairwiseAdonis)
library(tidyverse)
library(simboot)
library(multcomp)
library(hillR)
library(ggfortify)
library(picante)
library(seqinr)
library(MicrobiomeStat)
library(tibble)
library(magrittr)
library(ape)
library(ggpubr)
library(ggtext)
library(kableExtra)
set.seed(44)


samples <- read.csv("/Users/sonya.erlandson/OneDrive - USDA/greenhouse crop identity 2022/data tables/sampledata.csv", row.names = 1, stringsAsFactors = T)

maizesamp <- samples[samples$resp_spp=="maize", ]

soybeansamp <- samples[samples$resp_spp=="soybean", ]
```

```{r}

#how to use: need 1) otu table with samples as rows, taxa as columns
#                 2) a character vector of taxonomy level classifications that matches the taxa in otu table columns 
                        #-- should be the Kingdom, Phylum...etc column from taxonomy table.
# will return an otu table with taxa summed at the given taxonomic level.
# row names will be sample names and colnames will be same as taxonomy vector, plus one column of sample names for pivoting

aggregate_by_tax_level <- function(otu.table, taxonomy.vector){
  newOtuTable <- aggregate(t(otu.table), list(taxonomy.vector), FUN = sum)
  row.names(newOtuTable) <- newOtuTable$Group.1
  newOtuTable <- newOtuTable[ ,-1]
  newOtuTable <- as.data.frame(t(newOtuTable))
  
}
```


# 16S
Filter and tidy the bacterial data.
```{r}
### These are 99% OTUs with taxonomy assigned from the gg2 phylogeny/taxonomy
bacteria <- read.table("/Users/sonya.erlandson/OneDrive - USDA/greenhouse crop identity 2022/16S/greengenes2/otu_table_16S_gg2.tsv", sep = "\t", row.names = 1, header = T)
bacteria <- as.data.frame(t(bacteria))
#  read.table("/Users/sonya.erlandson/OneDrive - USDA/greenhouse crop identity 2022/16S/greengenes2/16S_otu_table_gg2.tsv", sep = "\t", header=T, row.names = 1)
bacteria.tax <- read.table("/Users/sonya.erlandson/OneDrive - USDA/greenhouse crop identity 2022/16S/greengenes2/taxonomy.tsv", sep = "\t", header = T)

bac.taxa.names <- bacteria.tax$Taxon
bac.taxa.names <- data.frame(gsub(pattern = "[dpocfgs]__", replacement = "", bac.taxa.names));colnames(bac.taxa.names) <- c("taxonomy") # remove taxonomy formatting
source("/Users/sonya.erlandson/OneDrive - USDA/greenhouse crop identity 2022/taxaLabels_for_gg2_16S.R")
bac.taxa.names <- separate(bac.taxa.names, col = taxonomy, into = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), sep = "; ")
bacteria.tax <- cbind(bacteria.tax, bac.taxa.names)
bacteria.tax$taxa_labels <- taxaLabels(bac.taxa.names)

summary(row.names(bacteria)==samples$idmatch)
bac.faith.pd <- read.table("/Users/sonya.erlandson/OneDrive - USDA/greenhouse crop identity 2022/16S/greengenes2/faiths_pd.tsv", header = T)
bac.unifrac <- read.table("/Users/sonya.erlandson/OneDrive - USDA/greenhouse crop identity 2022/16S/greengenes2/unifraq_weighted_dist_matrix.tsv")

# would need to match this up to taxonomic ids with the match table.
bac.norm <- read.csv("/Users/sonya.erlandson/OneDrive - USDA/greenhouse crop identity 2022/16S/EC_metagenome_out/seqtab_norm.tsv", sep = "\t") # 

bac.ec <- read.table("/Users/sonya.erlandson/OneDrive - USDA/greenhouse crop identity 2022/16S/EC_metagenome_out/pred_metagenome_unstrat.tsv", header = T, sep = "\t")

bac.path <- read.table("/Users/sonya.erlandson/OneDrive - USDA/greenhouse crop identity 2022/16S/pathways_out/path_abun_unstrat.tsv", header = T)
```

## 16S maize

subset maize data
```{r}
bac.maize.comm <- bacteria[samples$resp_spp=="maize", colSums(bacteria[samples$resp_spp=="maize", ])>0]

# drop ZmZm6 due to lower sample size (5123)
#which(row.names(bac.maize.comm)== "ZmZm6") 
bac.maize.comm <- bac.maize.comm[-16, ]
bac.maize.samp <- maizesamp[-16, ]

```

KEEP THIS ANALYSIS???
### species diversity
compare diversities of treatments. 
Does preceding species influence species richness and diversity?
```{r}
# calculate shannon or Hill's diversity
# comparisons maize = 1, pea = 2, soybean = 3, sunflower = 4
bac.maize.div <- mcpHill(dataf=data.frame(rrarefy(bac.maize.comm, sample = min(rowSums(bac.maize.comm)))), fact=factor(bac.maize.samp$cond_spp), boots=5000, qval=c(0,1,2), mattype = "Tukey") # 
bac.maize.div <- as.data.frame(bac.maize.div)
bac.maize.div$comparions <- c(rep(c("pea - maize", "soybean - maize", "sunflower - maize","soybean - pea", "sunflower - pea", "sunflower - soybean"),3))

```
This analysis compared all possible pairs of maize rhizospheres. 
May want to cut down comparisions?

Q=0 maize rhizosphere richness is different between pea vs maize,
soybean vs maize, sunflower vs maize, and soybean vs pea marginal). maize rhizosphere richness is not different between sunflower vs pea or sunflower vs soybean.

Q=1 maize rhizosphere diversity is different between pea vs maize, soybean vs maize, and sunflower vs maize. Soybean vs pea and sunflower vs pea are marginally different. Sunflower vs soybean are not different in maize rhizosphere diversity.

Q=2 maize rhizosphere diversity at q = 2 is significantly different only between sunflower vs pea and pea vs maize(p=0.004).

Overall, maize preceding maize decreases richness and diversity of bacteria in the maize rhizosphere compared to other crops. Pea preceding maize increases bacterial richness and diversity in the maize rhizosphere.

make plots:
```{r}
#need to calculate diversity at each level
bac.maize.div.table <- data.frame(cond_spp=factor(bac.maize.samp$cond_spp), 
                                  q0=hill_taxa(data.frame(rrarefy(bac.maize.comm, sample = min(rowSums(bac.maize.comm)))), q=0), 
                                  q1=hill_taxa(data.frame(rrarefy(bac.maize.comm, sample = min(rowSums(bac.maize.comm)))), q=1),
                                  q2=hill_taxa(data.frame(rrarefy(bac.maize.comm, sample = min(rowSums(bac.maize.comm)))), q=2))

ggplot(bac.maize.div.table)+
  geom_boxplot(aes(x = cond_spp, y = q0))+
  labs(x = "Preceding Species", y = "Effective Diversity: Hill *q* = 0", title = "Maize Rhizosphere Diversity")+
  theme_minimal()+
  theme(axis.text = element_markdown())
  stat_compare_means(aes( x = cond_spp, y = q0), method = "anova")
```

q = 1, inverse of shannon

```{r}
ggplot(bac.maize.div.table)+
  geom_boxplot(aes(x = cond_spp, y = q1))+
  labs(x = "Preceding Species", y = "Effective Diversity: Hill *q* = 1", title = "Maize Rhizosphere Diversity")+
  theme_minimal()+
  theme(axis.text = element_markdown())+
  stat_compare_means(aes( x = cond_spp, y = q1), method = "anova")
```

q = 2 inverse of simpson

```{r}
ggplot(bac.maize.div.table)+
  geom_boxplot(aes(x = cond_spp, y = q2))+
  labs(x = "Preceding Species", y = "Effective Diversity: Hill *q* = 2", title = "Maize Rhizosphere Diversity")+
  theme_minimal()+
  theme(axis.text = element_markdown())+
  stat_compare_means(aes( x = cond_spp, y = q2), method = "anova")
```

### phylogenetic diversity

Need a null model (in picante) to compare faith's pd. Could select another diversity metric if interested in how the phylogenetic structure of the community might reveal assembly of rhizosphere communities.

Faith's PD is a very basic phylogenetic metric. Need to get the phylogeny itself to do more interesting phylogenetic analyses
```{r}
# phylogenetic diversity - need faith's D  
# how to test? ANOVA?
bac.maize.phydiv <- bac.faith.pd[samples$resp_spp == "maize", ]

# normality and homogeneity
hist(bac.faith.pd[samples$resp_spp == "maize", ]$faith_pd, main = "bacteria phy div hist")
fligner.test(bac.faith.pd[samples$resp_spp == "maize", ]$faith_pd, maizesamp$cond_spp)
# can test with regular one way anova
# Compute the analysis of variance
maize.cond <- as.factor(maizesamp$cond_spp)
bac.maize.aov <- aov(bac.faith.pd[samples$resp_spp == "maize", ]$faith_pd ~ maize.cond)
# Summary of the analysis
summary(bac.maize.aov)
summary(glht(bac.maize.aov, linfct = mcp(maize.cond = "Tukey")))
```

Maize has the lowest phylogenetic diversity and increasing in soybean, sunflower, and highest in pea.

Maize rhizospheres preceded by pea, sunflower and soybean had higher phylogenetic diversity than maize (as preceding plant).

Pea was also higher than soybean, but not different than sunflower.

Plot
```{r}
phy.comparisons <- list( c("maize", "pea"), c("maize", "soybean"), c("maize", "sunflower"))

maize.cond <- factor(maize.cond, levels = c("maize", "soybean", "sunflower", "pea"), labels = c("Maize", "Soybean", "Sunflower", "Pea"))

ggplot(bac.faith.pd[samples$resp_spp == "maize", ], )+
  geom_boxplot(aes(x = maize.cond, y = faith_pd))+
  labs(x = "Preceding Species", y = "Faith's D")+
  theme_minimal()+
  stat_anova_test(aes(x = maize.cond, y = faith_pd))
# would need to add brackets
# Really just interested in a ranking
```

### Enzyme diversity
need a community otu table and a table of species and traits to calculate functional diversity using the hillR package. (species as rows, traits as columns)

get null data for statistical tests? there might not be enough reps (5)

```{r}
row.names(bac.ec) <- bac.ec$function.

bac.ec <- as.data.frame(t(bac.ec[,-1]))

maize.ec <- bac.ec[samples$resp_spp=="maize", colSums(bac.ec[samples$resp_spp=="maize", ])>0]

maize.ec.est.div <- mcpHill(dataf=maize.ec, fact=factor(maizesamp$cond_spp), boots=5000, qval=c(0,1), mattype = "Tukey") # 
maize.ec.est.div <- as.data.frame(maize.ec.est.div)
maize.ec.est.div$comparions <- rep(c("pea - maize", "soybean - maize", "sunflower - maize","soybean - pea", "sunflower - pea", "sunflower - soybean"),2)
maize.ec.est.div
```

make boxplot of enzyme richness
```{r}
bac.ec.maize.div <- data.frame(cond_spp=factor(maizesamp$cond_spp), 
                                  q0=hill_taxa(maize.ec, q=0))
ggplot(bac.ec.maize.div)+
  geom_boxplot(aes(x = cond_spp, q0))+
  labs(x = "preceding species", y = "Enzyme Richness - Hill q = 0", title = "maize rhizosphere richness, q = 0") # need to add anova differences
```

### Pathway diversity
```{r}
row.names(bac.path) <- bac.path$pathway

bac.path <- as.data.frame(t(bac.path[,-1]))

maize.path <- bac.path[samples$resp_spp=="maize", colSums(bac.path[samples$resp_spp=="maize", ])>0]

maize.path.est.div <- mcpHill(dataf=maize.path, fact=factor(maizesamp$cond_spp), boots=5000, qval=c(0), mattype = "Tukey") # 
maize.path.est.div <- as.data.frame(maize.path.est.div)
maize.path.est.div$comparions <- c("pea - maize", "soybean - maize", "sunflower - maize","soybean - pea", "sunflower - pea", "sunflower - soybean")
maize.path.est.div
```

make boxplot of pathway richness & diversity
```{r}
bac.path.maize.div <- data.frame(cond_spp=factor(maizesamp$cond_spp), 
                                  q0=hill_taxa(maize.ec, q=0),
                                 q1=hill_taxa(maize.ec, q=1))
ggplot(bac.path.maize.div)+
  geom_boxplot(aes(x = cond_spp, q0))+
  labs(x = "preceding species", y = "Pathway Richness - Hill q=0", title = "maize rhizosphere richness, q=0")

ggplot(bac.path.maize.div)+
  geom_boxplot(aes(x = cond_spp, q1))+
  labs(x = "preceding species", y = "Pathway Diversity - Hill q=1", title = "maize rhizosphere richness, q=1")# need to add anova differences
```

### beta diversity

Does preceding crop alter maize rhizosphere community composition? In
the field study, bacterial communities were differnt between rotations.

can look at unifraq only. 
```{r}
# permanova/rda to test difference between groups
# CLR transform community data.

# test
bac.maize.permanova <- adonis2(decostand(bac.maize.comm+1, method = "clr")~cond_spp, data = bac.maize.samp, method = "euclid", by = "terms")
bac.maize.pairedanova <- pairwise.adonis2(decostand(bac.maize.comm+1, method = "clr")~cond_spp, data = bac.maize.samp, method = "euclid")
bac.maize.permanova

lapply(bac.maize.pairedanova, FUN = kable)

#plot(prcomp(bac.maize.clr))
autoplot(prcomp(decostand(bac.maize.comm+1, method = "clr")), data = bac.maize.samp, color = 'cond_spp')+theme_minimal()
```

#### weighted unifrac

what's a better test here?
```{r}
autoplot(prcomp(bac.unifrac[samples$resp_spp=="maize", samples$resp_spp=="maize"]), color = 'cond_spp', data = maizesamp, )

anosim(bac.unifrac[samples$resp_spp=="maize", samples$resp_spp=="maize"], grouping = factor(maizesamp$cond_spp))
  
```

### differential abundance
less interested in differential abundance
Species level
Family level
Phylum level
```{r}

bac.maize.phylum <- aggregate_by_tax_level(bac.maize.comm, bacteria.tax[bacteria.tax$Feature.ID %in% colnames(bac.maize.comm), "phylum"])
bac.maize.family <- bac.maize.family[,-1]

bac.maize.samp$cond_spp <- as.factor(bac.maize.samp$cond_spp)

maize.bac.da <- linda(
feature.dat = t(bac.maize.phylum),
meta.dat = bac.maize.samp,
formula = "~cond_spp",
feature.dat.type = 'count',
prev.filter = 0.09,
adaptive = TRUE)


maize.bac.da.pea <- maize.bac.da$output$cond_spppea
# get species names
maize.bac.da.pea <- cbind(maize.bac.da.pea, bacteria.tax[match(row.names(maize.bac.da.pea), bacteria.tax$Feature.ID), c(5,11)])

maize.bac.da.soybean <- maize.bac.da$output$cond_sppsoybean

```

```{r}
linda.plot(maize.bac.da, variables.plot = "cond_sppsoybean", titles = "maize vs soybean (maize rhizosphere)")
# MAize is the baseline (A/B)
# These taxa are more abundant in the maize rhizosphere preceded by maize.
# could check to see if these are the same and compare to maarjam db

# Firmicutes are more abundance in maize, Myxococcota in soybean

```

```{r}

linda.plot(maize.bac.da, variables.plot = "cond_spppea", titles = "maize vs pea (maize rhizosphere)")
# LOTS more differentially abundant with pea

```

### ggpicrust2
not really interested in the specific functions...its all imaginary.
```{r}
library(ggh4x)
ecmeta <- read.table("/Users/sonya.erlandson/OneDrive - USDA/greenhouse crop identity 2022/16S/EC_metagenome_out/pred_metagenome_unstrat_descrip.tsv", sep = "\t", header = T)
normtab <- read.table("/Users/sonya.erlandson/OneDrive - USDA/greenhouse crop identity 2022/16S/EC_metagenome_out/seqtab_norm.tsv", sep = "\t", header = T)
kometa <- read.table("/Users/sonya.erlandson/OneDrive - USDA/greenhouse crop identity 2022/16S/KO_metagenome_out/pred_metagenome_unstrat.tsv", sep = "\t", header = T)

samples <- tibble(samples)
str(samples)

maize.ko <- kometa[,samples$resp_spp=="maize"]
maizesamp <- samples[samples$resp_spp=="maize",]

maize.kegg.out <- ggpicrust2(file = "/Users/sonya.erlandson/OneDrive - USDA/greenhouse crop identity 2022/16S/KO_metagenome_out/pred_metagenome_unstrat.tsv", 
                             metadata = samples, 
                             group = "cond_spp", 
                             pathway = "KO", daa_method = "LinDA",
                             select = samples$idmatch[samples$resp_spp=="maize"],
                             reference = "maize",ko_to_kegg = T, order = "pathway_class",
                                 p_values_bar = TRUE,
                                 x_lab = "pathway_name", method = , group1 = , group2 = )


kegg.abund <- ko2kegg_abundance(file = "/Users/sonya.erlandson/OneDrive - USDA/greenhouse crop identity 2022/16S/KO_metagenome_out/pred_metagenome_unstrat.tsv")
kegg.diff.abund <- pathway_daa(abundance = maize.kegg, metadata = maizesamp, group = "cond_spp",daa_method = "LinDA", reference = "maize")
pathway_pca(abundance = maize.kegg, metadata = maizesamp,group = "cond_spp")
pathway_heatmap(abundance = maize.kegg, metadata = maizesamp, group = "cond_spp")

```


###Network

Try SpeicEasi to construct networks. 
SparCC 
```{r}
library(devtools)
library(SpiecEasi)

library(microbiome) # data analysis and visualisation
library(RColorBrewer) # nice color options
library(ggpubr) # publication quality figures, based on ggplot2
library(dplyr) # data handling
library(network)
library(intergraph)
library(ggnet)
library(igraph)

bac.maize.maize <- spiec.easi(as.matrix(bac.maize.comm[bac.maize.samp$cond_spp == "maize", colSums(bac.maize.comm[bac.maize.samp$cond_spp == "maize", ])>0]), method='mb', sel.criterion = "stars", lambda.min.ratio=1e-2,
                          nlambda=20, pulsar.params=list(rep.num=50))

bac.maize.soybean <- spiec.easi(X, method='mb', lambda.min.ratio=1e-2, nlambda=15)
bac.maize.pea <- spiec.easi(X, method='mb', lambda.min.ratio=1e-2, nlambda=15)
```

## 16S soybean
subset soybean data

```{r}
bac.soybean.comm <- bacteria[samples$resp_spp=="soybean", colSums(bacteria[samples$resp_spp=="soybean", ])>0]
```

### species diversity

```{r}
# calculate shannon or Hill's diversity
# comparisons 1 = maize, 2 = soybean, 3 = spring wheat
bac.soybean.div <- mcpHill(dataf=data.frame(rrarefy(bac.soybean.comm, sample = min(rowSums(bac.soybean.comm)))), fact=factor(soybeansamp$cond_spp), boots=5000, qval=c(0,1,2), mattype = "Tukey") # 
bac.soybean.div <- as.data.frame(bac.soybean.div)
bac.soybean.div$comparions <- c(rep(c("soybean-maize", "spring wheat-soybean", "spring wheat - maize"),3))
bac.soybean.div
```

Q=0 soybean - maize are different, spring wheat-maize is marginal (0.05)

Q=1 ns Q=2 ns

q=0 plot, richness is different between soybean and maize.

```{r}
#need to calculate diversity at each level
bac.soybean.div.table <- data.frame(cond_spp=factor(soybeansamp$cond_spp), 
                                    q0=hill_taxa(data.frame(rrarefy(bac.soybean.comm, sample = min(rowSums(bac.soybean.comm)))), q=0), 
                                    q1=hill_taxa(data.frame(rrarefy(bac.soybean.comm, sample = min(rowSums(bac.soybean.comm)))), q=1),
                                    q2=hill_taxa(data.frame(rrarefy(bac.soybean.comm, sample = min(rowSums(bac.soybean.comm)))), q=2))
ggplot(bac.soybean.div.table)+
  geom_boxplot(aes(x = cond_spp, q0))+
  labs(x = "preceding species", y = "Effective Richness - Hill q=0", title = "soybean rhizosphere richness") # need to add anova differences
```

not significantly different

```{r}
ggplot(bac.soybean.div.table)+
  geom_boxplot(aes(x = cond_spp, q1))+
  labs(x = "preceding species", y = "Effective Diversity - Hill q=1") # need to add anova differences

ggplot(bac.soybean.div.table)+
  geom_boxplot(aes(x = cond_spp, q2))+
  labs(x = "preceding species", y = "Effective Diversity - Hill q=2") # need to add anova differences
```

### phylogenetic diversity

Faith's PD soybean rhizosphere phylogenetic diversity is different
between pairs, except sunflower-soybean and sunflower - pea. Soybean -
soybean is marginal.

```{r}
# phylogenetic diversity - need faith's D  
#bac.soybean.phydiv <- bac.faith.pd[samples$resp_spp == "soybean", ]
summary(bac.faith.pd[samples$resp_spp == "soybean",]$SampleID == soybeansamp$idmatch)
# normality and homogeneity
hist(bac.faith.pd[samples$resp_spp == "soybean", ]$faith_pd)
fligner.test(bac.faith.pd[samples$resp_spp == "soybean", ]$faith_pd, soybeansamp$cond_spp)
# can test with regular one way anova
# Compute the analysis of variance
soybean.cond <- as.factor(soybeansamp$cond_spp)
bac.soybean.aov <- aov(bac.faith.pd[samples$resp_spp == "soybean",2] ~ soybean.cond)
# Summary of the analysis
summary(bac.soybean.aov)
summary(glht(bac.soybean.aov, linfct = mcp(soybean.cond = "Tukey")))
```

faiths pd plot

```{r}
ggplot(bac.faith.pd[samples$resp_spp == "soybean", ])+
  geom_boxplot(aes(x = soybean.cond, faith_pd))+
  labs(x = "preceding species", y = "Faith's D", title = "soybean rhizosphere phylogenetic diversity") # need to add anova differences
  
```

### beta diversity

Does preceding crop alter soybean rhizosphere community composition? in
the field study the bacterial rhizosphere communities did NOT differ
between rotations.

```{r}
# permanova/rda to test difference between groups
# CLR transform community data.

# test
bac.soybean.permanova <- adonis2(decostand(bac.soybean.comm+1, method = "clr")~cond_spp, data = soybeansamp, method = "euclid", by = "terms")
bac.soybean.pairedanova <- pairwise.adonis2(decostand(bac.soybean.comm+1, method = "clr")~cond_spp, data = soybeansamp, method = "euclid")
bac.soybean.permanova
bac.soybean.pairedanova 
#plot(prcomp(bac.soybean.clr))
autoplot(prcomp(bac.soybean.clr), data = soybeansamp, color = 'cond_spp') 
```


### weighted unifrac - need 3D to see all factors

```{r}
autoplot(prcomp(bac.unifrac[samples$resp_spp=="soybean", samples$resp_spp=="soybean"]), color = 'cond_spp', data = soybeansamp)

anosim(bac.unifrac[samples$resp_spp=="soybean", samples$resp_spp=="soybean"], grouping = factor(soybeansamp$cond_spp))

  
```

# ITS1

```{r}
### These are 97% OTUs with taxonomy assigned from unite with sintax in vsearch. bootstrap cut off was 0.60.
fungi <- read.table("/Users/sonya.erlandson/OneDrive - USDA/greenhouse crop identity 2022/ITS/otu_table_its1_97.txt", sep = "\t", header=T, row.names = 1)

all.tax <- read.table("/Users/sonya.erlandson/OneDrive - USDA/greenhouse crop identity 2022/ITS/taxonomy_otus.txt", sep = "\t", header = F)

# fix taxonomy file formatting
all.tax <- separate(all.tax, col = V1, into = c("otu_id", "size"), sep = ";size=")
all.tax <- all.tax[,-4] # remove this extra column
all.tax$size <- as.numeric(all.tax$size)#make size numeric for sorting
all.tax <- all.tax[order(all.tax$size, decreasing = T), ] # order by size
all.tax$taxonomy <- gsub(pattern = "[dpocfgs]:", replacement = "", all.tax$V4 )# remove taxonomy formatting

all.tax$taxonomy_sep <- gsub(pattern = "[dpocfgs]:", replacement = "", all.tax$V4 ) # make a column for separating into each level
#all.tax$taxonomy_sep <- str_remove_all(all.tax$taxonomy_sep, pattern = "\\(....\\)")
all.tax <- separate(all.tax, col = taxonomy_sep, into = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), sep = ",") # make each their own column

# remove plants
fungi.tax <- all.tax[str_detect(pattern = "Viridiplantae", string = all.tax$V2, negate = T) & as.numeric(all.tax$size) >1 , ] # get fungi with size >=2
# add taxa labels
source("/Users/sonya.erlandson/OneDrive - USDA/greenhouse crop identity 2022/taxaLabels_species_level.R")
fungi.tax$taxa_labels <- taxaLabels(fungi.tax)
# remove NAs
fungi.tax[is.na(fungi.tax)] <- ""

# reorder taxa in otu table by size to match tax table
fungi <- fungi[match(all.tax$otu_id, row.names(fungi)), ] # order by size
summary(row.names(fungi)==all.tax$otu_id)

fungi <- as.data.frame(t(fungi[,-1])) # transform otu table and remove extraction nc
fungi <- as.data.frame(fungi[match(samples$idmatch, row.names(fungi)), ]) # get samples and otu table in the same order.
fungi <- fungi[,colnames(fungi) %in% fungi.tax$otu_id]


# make table for funguild
# funguild <- as.data.frame(t(fungi))
# funguild$taxonomy <- fungi.tax$taxonomy
# 
# write.table(funguild, "funguild_otu_table.txt", quote = F, sep = "\t")
# 
# test <- read.table("/Users/sonya.erlandson/Documents/funguild_otu_table.txt", sep = "\t")

funguild <- read.table("/Users/sonya.erlandson/OneDrive - USDA/greenhouse crop identity 2022/ITS/funguild_otu_table.guilds_matched.txt", sep = "\t", header = T)

```

## ITS maize

subset maize data

```{r}
fungi.maize.comm <- fungi[samples$resp_spp=="maize", colSums(fungi[samples$resp_spp=="maize", ])>0]
```


### species diversity
Potentially skip richness and diversity and just focus on phylogenetic and taxonomic diversity. 

```{r}
# calculate shannon or Hill's diversity
# comparisons maize = 1, pea = 2, soybean = 3, sunflower = 4
fungi.maize.div <- mcpHill(dataf=data.frame(rrarefy(fungi.maize.comm, sample = min(rowSums(fungi.maize.comm)))), fact=factor(maizesamp$cond_spp), boots=5000, qval=c(0,1,2), mattype = "Tukey") # 
fungi.maize.div <- as.data.frame(fungi.maize.div)
fungi.maize.div$comparions <- c(rep(c("pea - maize", "soybean - maize", "sunflower - maize","soybean - pea", "sunflower - pea", "sunflower - soybean"),3))

```

calculate diversity metrics for each preceding crop treatment
```{r}
#need to calculate diversity at each level
fungi.maize.div.table <- data.frame(cond_spp=factor(maizesamp$cond_spp),
                                    q0=hill_taxa(rrarefy(fungi.maize.comm, sample = min(rowSums(fungi.maize.comm))), q=0),
                                    q1=hill_taxa(rrarefy(fungi.maize.comm, sample = min(rowSums(fungi.maize.comm))), q=1),
                                    q2=hill_taxa(rrarefy(fungi.maize.comm, sample = min(rowSums(fungi.maize.comm))), q=2))
```

plot diversity metrics 
q=0
```{r}
ggplot(fungi.maize.div.table)+
  geom_boxplot(aes(x = cond_spp, q0))+
  labs(x = "preceding species", y = "Effective Taxa Richness - Hill q=0") # need to add anova differences
```

q = 1

```{r}
ggplot(fungi.maize.div.table)+
  geom_boxplot(aes(x = cond_spp, q1))+
  labs(x = "preceding species", y = "Effective Diversity - Hill q=1") # need to add anova differences
```

q = 2

```{r}
ggplot(fungi.maize.div.table)+
  geom_boxplot(aes(x = cond_spp, q2))+
  labs(x = "preceding species", y = "Effective Diversity - Hill q=2") # need to add anova differences
```


### taxonomic distinctiveness
using the vegan functions to test taxonomic diversity (D) and distinctiveness (D*).
Also calculates variability in taxonomic distinctiveness (Lambda)
expect greater taxonomic distinctiveness in heterospecific preceding crops and lower in the conspecific preceding crop.
```{r}
maize.tax <- fungi.tax[fungi.tax$otu_id %in% colnames(fungi.maize.comm),]

# taxonomic distance and distinctiveness
# check order
summary(colnames(fungi.maize.comm)== maize.tax$otu_id)

# calculate distance matrix
tax.dist.mat <- taxa2dist(maize.tax[,6:12], varstep = TRUE, check = TRUE)
# calculate D metrics
test.taxon.div <- taxondive(decostand(fungi.maize.comm+1, method = "clr"), tax.dist.mat, match.force = FALSE)

# interested in delta+ (taxonomic distinctiveness) and lambda (variance in taxonomic distinctiveness)
hist(test.taxon.div$Dplus)

maize.tax.dist.aov <- aov(formula = test.taxon.div$Dplus ~ maizesamp$cond_spp)
summary(maize.tax.dist.aov)
summary(glht(maize.tax.dist.aov, linfct = mcp("maizesamp$cond_spp" = "Tukey")))
ggplot()+
  geom_boxplot(aes(test.taxon.div$Dplus, x = maizesamp$cond_spp))+
  labs(x = "conditioning species", y ="delta+ - taxonomic distinctiveness", title = "maize rhizosphere fungi")
```

variation in taxonomic distinctiveness
```{r}
hist(test.taxon.div$Lambda)
maize.tax.lambda.aov <- aov(formula = test.taxon.div$Lambda ~ maizesamp$cond_spp)
summary(maize.tax.lambda.aov)
summary(glht(maize.tax.lambda.aov, linfct = mcp("maizesamp$cond_spp" = "Tukey")))
ggplot()+
  geom_boxplot(aes(test.taxon.div$Lambda, x = maizesamp$cond_spp))+
  labs(x = "conditioning species", y ="Lambda - variance in taxonomic distinctiveness", title = "maize rhizosphere fungi")
```

### Guild Diversity
to use functional diversity metrics need 1)otu table 2) table of species traits
Could make a presence-absence guild table that looks like this:
    guild1    guild2    guild3
sp1 1           0       1
sp2 0           0       1


```{r}
colnames(funguild)
unique(funguild$Guild)
write.csv(funguild[,-72], "funguild.csv", quote = F)

funguild <- funguild[match(fungi.tax$otu_id[fungi.tax$otu_id %in% funguild$OTU_ID], funguild$OTU_ID), ]

funguild.otus <- funguild[,1:71]
funguild.guilds <- separate(funguild[,c(1,76)], col = "Guild", sep = "-", into = c("a","b","c","d","e","f","g","h"))


### approach ONE. with out functional diversity. 

# need to do some fancy magrittr 
# combine levels of the same group
# subset to just groups
maize.guild <- as.data.frame(funguild[,c(76,2:71)])

maize.guild <- as.data.frame(aggregate(.~Guild, FUN = sum, data = maize.guild ))

row.names(maize.guild) <- maize.guild$Guild
maize.guild <- as.data.frame(t(maize.guild[,-1]))
maize.guild <- sapply(maize.guild, FUN = function(x) as.numeric(x))

maize.guild <- as.data.frame(maize.guild[samples$resp_spp=="maize", colSums(maize.guild[samples$resp_spp=="maize", ])>0])

row.names(maize.guild) <- maizesamp$idmatch

maize.guild.div <- mcpHill(dataf=maize.guild, fact=factor(maizesamp$cond_spp), boots=5000, qval=c(0,1), mattype = "Tukey") # 
maize.guild.div <- as.data.frame(maize.guild.div)
maize.guild.div$comparions <- c("pea - maize", "soybean - maize", "sunflower - maize","soybean - pea", "sunflower - pea", "sunflower - soybean")



maize.guild.div.table <- data.frame(cond_spp=factor(maizesamp$cond_spp),
                                  q0=hill_taxa(maize.guild, q=0),
                                  q1=hill_taxa(maize.guild, q=1))
ggplot(maize.guild.div.table)+
  geom_boxplot(aes(x = cond_spp, q0))+
  labs(x = "preceding species", y = "Effective Richness - Hill q=0", title = "maize rhizosphere richness, q=0") # need to add anova differences


ggplot(maize.guild.div.table)+
  geom_boxplot(aes(x = cond_spp, q1))+
  labs(x = "preceding species", y = "Effective Diversity - Hill q=0", title = "maize rhizosphere richness, q=1")

```

### beta diversity

Does preceding crop alter maize rhizosphere community composition?

```{r}
# permanova/rda to test difference between groups
# CLR transform community data.

# test
fungi.maize.permanova <- adonis2(decostand(fungi.maize.comm+1, method = "clr")~cond_spp, data = maizesamp, method = "euclid", by = "terms")
fungi.maize.pairedanova <- pairwise.adonis2(decostand(fungi.maize.comm+1, method = "clr")~cond_spp, data = maizesamp, method = "euclid")
fungi.maize.permanova
fungi.maize.pairedanova 
#plot(prcomp(fungi.maize.clr))
autoplot(prcomp(decostand(fungi.maize.comm+1, method = "clr")), data = maizesamp, color = 'cond_spp') 
```

###AMF community

expect the AMF community to be different among the treatments,
especially soybean vs maize. -in the field study, the AMF communities were different among the rotations at the seedling stage, but not between soy
vs. other preceding crops or at flowering. (not surprising)

```{r}
# get AMF community subset. Test for differences among treatments. Do differential abundance?
maize.amf <- fungi.maize.comm[ ,colnames(fungi.maize.comm) %in% fungi.tax$otu_id[str_detect(string = fungi.tax$phylum, pattern = "Glomeromycota")]]

adonis2(maize.amf~cond_spp, data = maizesamp)
pairwise.adonis(maize.amf, as.factor(maizesamp$cond_spp))
```

community pairs that are significantly different:
***sunflower - soybean***
**sunflower - pea**
***maize - soybean***
**maize - pea**
***soybean - pea***



### differential abundance

```{r}
summary(colSums(decostand(maize.amf, method = "pa"))>4)
maizesamp$cond_spp <- as.factor(maizesamp$cond_spp)

maize.amf.da <- linda(
feature.dat = t(maize.amf),
meta.dat = maizesamp,
formula = "~cond_spp",
feature.dat.type = 'count',
prev.filter = 0.09,
adaptive = TRUE)

fungi.maize.family <- aggregate_by_tax_level(fungi.maize.comm, fungi.tax[fungi.tax$otu_id %in% colnames(fungi.maize.comm), "family"])

maize.fungi.family.da <- linda(
feature.dat = t(fungi.maize.family),
meta.dat = maizesamp,
formula = "~cond_spp",
feature.dat.type = 'count',
prev.filter = 0.09,
adaptive = TRUE)



fungi.tax[match(row.names(maize.amf.da$output$cond_spppea), fungi.tax$otu_id), "taxa_labels"]
```

```{r}
linda.plot(maize.fungi.family.da, variables.plot = "cond_sppsoybean", titles = "maize vs soybean (maize rhizosphere)")
# MAize is the baseline (A/B)
# These taxa are more abundant in the maize rhizosphere preceded by maize.
# could check to see if these are the same and compare to maarjam db
```

```{r}

linda.plot(maize.fungi.family.da, variables.plot = "cond_spppea", titles = "maize vs pea (maize rhizosphere)")
# LOTS more differentially abundant with pea - many AMF taxa are less abundant in pea-conditioned treatments than maize-conditioned treatments.

# only two are more abundant in pea-conditioned maize rhizospheres
fungi.tax[which(fungi.tax$otu_id=="OTU_245"), 5] # glomeraceae
fungi.tax[which(fungi.tax$otu_id=="OTU_79"), 5] # septoglomus

```

## ITS soybean

subset soybean data

```{r}
fungi.soybean.comm <- fungi[samples$resp_spp=="soybean", colSums(fungi[samples$resp_spp=="soybean", ])>0]
#fungi.soybeanra <- fungi.soybean.comm/rowSums(fungi.soybean.comm)

#fungi.soybean.clr <- decostand(fungi.soybean.comm+1, method = "clr")
#fungi.soybean.rar <- data.frame(rrarefy(fungi.soybean.comm, sample = min(rowSums(fungi.soybean.comm))))
```

proportion of taxa that are in all samples, 90% and 80%: most taxa are
rare

```{r}
fungi.soybean.pa <- decostand(fungi.soybean.comm, method = "pa")
hist(colSums(fungi.soybean.pa))
30*.9
summary(colSums(fungi.soybean.pa)==30) # 109 taxa, 0.034 of all taxa 
summary(colSums(fungi.soybean.pa)>=27) # 214, 0.07 of all taxa
summary(colSums(fungi.soybean.pa)>=24) # 295 0.09 of all taxa 
295/3248

# rank abundance plots?


```

### species diversity

```{r}
# calculate shannon or Hill's diversity
# comparisons 1 = maize, 2 = soybean, 3 = spring wheat
fungi.soybean.div <- mcpHill(dataf=data.frame(rrarefy(fungi.soybean.comm, sample = min(rowSums(fungi.soybean.comm)))), fact=factor(soybeansamp$cond_spp), boots=5000, qval=c(0,1,2), mattype = "Tukey") # 
fungi.soybean.div <- as.data.frame(fungi.soybean.div)
fungi.soybean.div$comparions <- c(rep(c("soybean-maize", "spring wheat-soybean", "spring wheat - maize"),3))

```

```{r}
#need to calculate diversity at each level
fungi.soybean.div.table <- data.frame(cond_spp=factor(soybeansamp$cond_spp), 
                                      q0=hill_taxa(data.frame(rrarefy(fungi.soybean.comm, sample = min(rowSums(fungi.soybean.comm)))), q=0), 
                                      q1=hill_taxa(data.frame(rrarefy(fungi.soybean.comm, sample = min(rowSums(fungi.soybean.comm)))), q=1),
                                      q2=hill_taxa(data.frame(rrarefy(fungi.soybean.comm, sample = min(rowSums(fungi.soybean.comm)))), q=2))
ggplot(fungi.soybean.div.table)+
  geom_boxplot(aes(x = cond_spp, q0))+
  labs(x = "preceding species", y = "Effective Richness - Hill q=0") # need to add anova differences
```

```{r}
ggplot(fungi.soybean.div.table)+
  geom_boxplot(aes(x = cond_spp, q1))+
  labs(x = "preceding species", y = "Effective Diversity - Hill q=1") # need to add anova differences
```

```{r}
ggplot(fungi.soybean.div.table)+
  geom_boxplot(aes(x = cond_spp, q2))+
  labs(x = "preceding species", y = "Effective Diversity - Hill q=2") # need to add anova differences
```

### permanovas

Does preceding crop alter soybean rhizosphere community composition?

```{r}
# permanova/rda to test difference between groups
# CLR transform community data.

# test
fungi.soybean.permanova <- adonis2(decostand(fungi.soybean.comm+1, method = "clr")~cond_spp, data = soybeansamp, method = "euclid", by = "terms")
fungi.soybean.pairedanova <- pairwise.adonis2(decostand(fungi.soybean.comm+1, method = "clr")~cond_spp, data = soybeansamp, method = "euclid")
fungi.soybean.permanova
fungi.soybean.pairedanova 
#plot(prcomp(fungi.soybean.clr))
autoplot(prcomp(decostand(fungi.soybean.comm+1, method = "clr")), data = soybeansamp, color = 'cond_spp') 
```

### AMF community

in field study, AMF were different with soybean preceding maize.

```{r}
# get AMF community subset. Test for differences among treatments. Do differential abundance?
soybean.amf <- fungi.soybean.comm[ ,colnames(fungi.soybean.comm) %in% fungi.tax$otu_id[str_detect(string = fungi.tax$phylum, pattern = "Glomeromycota")]]

adonis2(soybean.amf~cond_spp, data = soybeansamp)
pairwise.adonis(soybean.amf, as.factor(soybeansamp$cond_spp))

soybeansamp$cond_spp <- factor(soybeansamp$cond_spp, levels = c("soybean", "maize", "spring wheat"))
soybean.linda.amf <- linda(t(soybean.amf), 
                           soybeansamp, 
                           formula = "~cond_spp", 
                           feature.dat.type = "count", 
                           zero.handling = "pseudocount",
                           adaptive = T,
                           prev.filter = 0.09)
```

```{r}
linda.plot(soybean.linda.amf, variables.plot = c("cond_sppmaize"))
```

```{r}
# These are more abundant in soy rhizospheres when preceded by corn:
# similar taxonomic groups that are more abundant in maize when preceded by 
fungi.tax[which(fungi.tax$otu_id=="OTU_258"), 5]# Diversispora
fungi.tax[which(fungi.tax$otu_id=="OTU_179"), 5] # # no ID
fungi.tax[which(fungi.tax$otu_id=="OTU_1473"), 5] # paraglomerales
fungi.tax[which(fungi.tax$otu_id=="OTU_2386"), 5] # paraglomerales

```

```{r}
linda.plot(soybean.linda.amf, variables.plot = c("cond_sppspring wheat"))

```

differential otus

```{r}
fungi.tax[which(fungi.tax$otu_id=="OTU_258"), 5:10]
fungi.tax[which(fungi.tax$otu_id=="OTU_2386"), 5:10]
fungi.tax[which(fungi.tax$otu_id=="OTU_176"), 5:10]
fungi.tax[which(fungi.tax$otu_id=="OTU_1473"), 5:10]

# Paraglomerales and Diversispora
```

# Co-occurrence Analysis
SpiecEasi now includes a convenience wrapper for dealing with multiple taxa sequenced on the same samples, such as 16S and ITS, as seen in Tipton, Müller, et. al. (2018). It assumes that each taxa is in it’s own data matrix and that all samples are in all data matrices in the same order.

Here’s an example run from the HMP2 project with 16S and Proteomics data.
```{r}

se.hmp2 <- spiec.easi(list(hmp216S, hmp2prot), method='mb', nlambda=40,
              lambda.min.ratio=1e-2, pulsar.params = list(thresh = 0.05))

dtype <- c(rep(1,ntaxa(hmp216S)), rep(2,ntaxa(hmp2prot)))
plot(adj2igraph(getRefit(se.hmp2)), vertex.color=dtype+1, vertex.size=9)

```


load data and packages

```{r}
library(multcomp)
library(emmeans)
library(tidyverse)


samp <- read.csv("../../sonya.erlandson/OneDrive - USDA/greenhouse crop identity 2022/data tables/sampledata.csv", header = T)

maize <- samp[samp$resp_spp=="maize", ]
soy <- samp[samp$resp_spp=="soybean", ]
```

#Biomass
## maize
have 
-conditioning spp biomass
-maize or soy spp biomass
-germination date # what would this mean?
-maize and soy spad # should correlate with N
-growth stage # what would this mean?
```{r}
hist(maize$response.dry.mass.g)
maizelm <- lm(maize$response.dry.mass.g ~ maize$cond_spp)
#anova(lm(maize$response.dry.mass.g ~ maize$cond_spp))
maizeemmeans <- emmeans(maizelm, ~ cond_spp)
pwpm(maizeemmeans)
cld(maizeemmeans)
```


```{r}

ggplot(maize)+
  geom_boxplot(aes(y=response.dry.mass.g, x= cond_spp))+
  labs(x = "preceding species", y = "Maize dry biomass (grams)")

```

## soybean
```{r}
hist(soy$response.dry.mass.g)
soylm <- lm(soy$response.dry.mass.g ~ soy$cond_spp)
#anova(soylm)
soyemmeans <- emmeans(soylm, ~ cond_spp)
pwpm(soyemmeans)
cld(soyemmeans)
```

```{r}
ggplot(soy)+
  geom_boxplot(aes(y=response.dry.mass.g, x= cond_spp))+
  labs(y = "Soybean dry biomass (grams)", x = "preceding species")

```
#SPAD
## maize
```{r}
hist(maize$SPAD.mean3)
maizelm <- lm(maize$SPAD.mean3 ~ maize$cond_spp)

#anova(lm(maize$SPAD.mean3 ~ maize$cond_spp))

maizeemmeans <- emmeans(maizelm, ~ cond_spp)
pwpm(maizeemmeans)
cld(maizeemmeans)
```
```{r}
ggplot(maize)+
  geom_boxplot(aes(y=SPAD.mean3, x= cond_spp))+
  labs(x = "preceding species", y = "Maize SPAD")

```

## soybean
```{r}
hist(soy$SPAD.mean3)
soylm <- lm(soy$SPAD.mean3 ~ soy$cond_spp)
#anova(soylm)
soyemmeans <- emmeans(soylm, ~ cond_spp)
pwpm(soyemmeans)
cld(soyemmeans)

```

```{r}
ggplot(soy)+
  geom_boxplot(aes(y=SPAD.mean3, x= cond_spp))+
  labs(x = "preceding species", y = "Soybean SPAD")

```
